feat: create local proxy library that is Go Mobile-friendly (#64)
Jigsaw-Code/outline-sdk: x/examples/mobileproxy/README.md

Author: What do you think about moving this to x/appproxy  and call it x/mobileproxy, so it's clear it's for Go Mobile?

x/appproxy shouldn't really be used by Go code, since it has a limited API that is not really optimal for Go.

Author: FYI, I updated the PR to have everything in `x/mobileproxy`. This way we can keep the library as clearly intended for Go Mobile.

# Local Proxy Library for Mobile Apps

> This package illustrates the use Go Mobile to generarate a mobile library to run a local proxy and configure your app networking libraries.

Reviewer: ```suggestion
This package illustrates the use of Gomobile to generate a mobile library to run a local proxy and configure your app networking libraries.
```

Author: Fixed

> ### Build the Go Mobile binaries with [`go build`](https://pkg.go.dev/cmd/go#hdr-Compile_packages_and_dependencies)

Reviewer: ```suggestion
### Build the Gomobile binaries with [`go build`](https://pkg.go.dev/cmd/go#hdr-Compile_packages_and_dependencies)
```

Author: Keeping as Go Mobile as mentioned before.

```bash
go build -o ./out/ golang.org/x/mobile/cmd/gomobile golang.org/x/mobile/cmd/gobind
```

### Build the iOS and Android libraries with [`gomobile bind`](https://pkg.go.dev/golang.org/x/mobile/cmd/gomobile#hdr-Build_a_library_for_Android_and_iOS)

```bash

> PATH="$PATH:$(pwd)/out" $(pwd)/out/gomobile bind -target=ios -o "$(pwd)/out/LocalProxy.xcframework" github.com/Jigsaw-Code/outline-sdk/x/appproxy

Reviewer: Since we already set the `PATH`, this is redundant?

```suggestion
PATH="$PATH:$(pwd)/out" gomobile bind -target=ios -o "$(pwd)/out/LocalProxy.xcframework" github.com/Jigsaw-Code/outline-sdk/x/appproxy
```

Author: Good point, fixed.
I switched the PATH order to make sure we use the gomobile in the PATH, not from the system-level install.
